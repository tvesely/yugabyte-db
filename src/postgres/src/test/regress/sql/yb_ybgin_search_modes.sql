--
-- Yugabyte-owned test for ybgin index access method and search modes.
--

-- Disable sequential scan so that index scan is always chosen.
SET enable_seqscan = off;

CREATE INDEX NONCONCURRENTLY idx_partial ON arrays
    USING ybgin (a)
    WHERE a <@ '{1}' or a @> '{}' or a is null;

-- GIN_SEARCH_MODE_INCLUDE_EMPTY
SELECT * FROM arrays WHERE a <@ '{1}';
-- GIN_SEARCH_MODE_ALL
SELECT * FROM arrays WHERE a @> '{}';
-- GIN_CAT_NULL_ITEM
SELECT * FROM arrays WHERE a is null;

-- Cleanup
DROP INDEX idx_partial;
