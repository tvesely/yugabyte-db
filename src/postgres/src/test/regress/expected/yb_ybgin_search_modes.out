--
-- Yugabyte-owned test for ybgin index access method and search modes.
--
-- Disable sequential scan so that index scan is always chosen.
SET enable_seqscan = off;
CREATE INDEX NONCONCURRENTLY idx_partial ON arrays
    USING ybgin (a)
    WHERE a <@ '{1}' or a @> '{}' or a is null;
-- GIN_SEARCH_MODE_INCLUDE_EMPTY
SELECT * FROM arrays WHERE a <@ '{1}';
ERROR:  unsupported ybgin index scan
DETAIL:  ybgin index method does not support non-default search mode: include-empty.
-- GIN_SEARCH_MODE_ALL
SELECT * FROM arrays WHERE a @> '{}';
ERROR:  unsupported ybgin index scan
DETAIL:  ybgin index method does not support non-default search mode: all.
-- GIN_CAT_NULL_ITEM
SELECT * FROM arrays WHERE a is null;
ERROR:  unsupported ybgin index scan
DETAIL:  ybgin index method does not support non-default search mode: everything.
-- Cleanup
DROP INDEX idx_partial;
